src/deno_dir.rs

```diff
@@ -136,38 +137,81 @@ impl DenoDir {
   }
 
   // Prototype https://github.com/denoland/deno/blob/golang/deno_dir.go#L37-L73
+  /// Fetch remote source code.
   fn fetch_remote_source(
     self: &Self,
     module_name: &str,
     filename: &str,
-  ) -> DenoResult<(String, msg::MediaType)> {
-    let p = Path::new(filename);
-    // We write a special ".mime" file into the `.deno/deps` directory along side the
-    // cached file, containing just the media type.
-    let mut media_type_filename = filename.to_string();
-    media_type_filename.push_str(".mime");
-    let mt = Path::new(&media_type_filename);
-
-    let src = if self.reload || !p.exists() {
-      eprintln!("Downloading {}", module_name);
-      let (source, content_type) = http_util::fetch_sync_string(module_name)?;
-      match p.parent() {
-        Some(ref parent) => fs::create_dir_all(parent),
-        None => Ok(()),
-      }?;
-      deno_fs::write_file(&p, source.as_bytes(), 0o666)?;
-      deno_fs::write_file(&mt, content_type.as_bytes(), 0o666)?;
-      (source, map_content_type(&p, Some(&content_type)))
-    } else {
-      let source = fs::read_to_string(&p)?;
-      // .mime file might not exists with bundled deps
+  ) -> DenoResult<Option<CodeFetchOutput>> {
+    let extensions = ["", ".ts", ".js"];
+    for ext in extensions.iter() {
+      let filename = format!("{}{}", filename, ext);
```

Comment:
Do not use `format!` for concatenating strings. This is considered an [anti-pattern](https://github.com/rust-unofficial/patterns/blob/master/idioms/concat-format.md).

Reply:
There are several ways to do this. Please refer to the benchmark result https://github.com/hoodie/concatenation_benchmarks-rs

Reply:
Thanks for pointing out. I'll go with `array_concat` as it seems simple enough and fast